test/extended/util/test.go

```diff
@@ -248,126 +262,124 @@ func createTestingNS(baseName string, c kclientset.Interface, labels map[string]
 }
 
 var (
+	testMaps = map[string][]string{
+		// tests that require a local host
+		"[Local]": {
+			// Doesn't work on scaled up clusters
+			`\[Feature:ImagePrune\]`,
+		},
+		// alpha features that are not gated
+		"[Disabled:Alpha]": {
+			`\[Feature:Initializers\]`,           // admission controller disabled
+			`\[Feature:LocalPersistentVolumes\]`, // flag gate is off
+			`\[Feature:PodPreemption\]`,          // flag gate is off
+			`AdmissionWebhook`,                   // needs to be enabled
+		},
+		// tests for features that are not implemented in openshift
+		"[Disabled:Unimplemented]": {
+			`\[Feature:Networking-IPv6\]`, // openshift-sdn doesn't support yet
+			`Monitoring`,                  // Not installed, should be
+			`Cluster level logging`,       // Not installed yet
+			`Kibana`,                      // Not installed
+			`Ubernetes`,                   // Can't set zone labels today
+			`kube-ui`,                     // Not installed by default
+			`^Kubernetes Dashboard`,       // Not installed by default (also probably slow image pull)
+			`Ingress`,                     // Not enabled yet
+
+			`NetworkPolicy between server and client should allow egress access on one named port`, // not yet implemented
+
+			`should proxy to cadvisor`, // we don't expose cAdvisor port directly for security reasons
+		},
+		// tests that rely on special configuration that we do not yet support
+		"[Disabled:SpecialConfig]": {
+			`\[Feature:ImageQuota\]`,                    // Quota isn't turned on by default, we should do that and then reenable these tests
+			`\[Feature:Audit\]`,                         // Needs special configuration
+			`\[Feature:LocalStorageCapacityIsolation\]`, // relies on a separate daemonset?
+
+			`kube-dns-autoscaler`,                                                    // Don't run kube-dns
+			`should check if Kubernetes master services is included in cluster-info`, // Don't run kube-dns
+			`DNS configMap`, // this tests dns federation configuration via configmap, which we don't support yet
+
+			// vSphere tests can be skipped generally
+			`vsphere`,
+			`Cinder`, // requires an OpenStack cluster
+			// See the CanSupport implementation in upstream to determine wether these work.
+			`Ceph RBD`,                              // Works if ceph-common Binary installed (but we can't guarantee this on all clusters).
+			`GlusterFS`,                             // May work if /sbin/mount.glusterfs to be installed for plugin to work (also possibly blocked by serial pulling)
```

Comment:
I don't understand how the list of unsupported volume tests is created.

* AWS and GCE are not listed, while OpenStack Cinder is. Why can't anyone run the tests on OpenStack? BTW, all storage test should have proper `framework.SkipUnlessProviderIs("xyz")` and both `vsphere` and `cinder` should be included in the test (i.e. excluded from this list), at least when it comes to storage tests.

* Gluster and Ceph RBD are listed, while CephFS and NFS and iSCSI is not. All of these need something installed on the host. Note that these tests are *not* skipped automatically if appropriate tools are missing (iscsiadm, /bin/rbd, mount.glusterfs...)You should skip either all of them or none. There is a possibility to run these dependencies in containers, so the tests could eventually run everywhere, but it needs a separate test suite that runs these containers and enables appropriate feature. For now, I would add to the skip list at least `SCSI`, `Ceph` (that will match both CephFS and Ceph RBD)

Reply:
> Why can't anyone run the tests on OpenStack? 

Because the openstack tests hard failed rather than skipping gracefully (at the time they were added to the list)

> Note that these tests are not skipped automatically if appropriate tools are missing (iscsiadm, /bin/rbd, mount.glusterfs...)

If the test does this, then we're good.  But that's not how the tests are behaving today.

Only tests that have failed have made it into the list.  So any test that now skips correctly I will be happy to remove the exclusions for.

Reply:
> Only tests that have failed have made it into the list. So any test that now skips correctly I will be happy to remove the exclusions for.

They failed on one particular RHEL installation since it had e.g. iscsiadm installed but not /bin/rbd. They will fail on other distros (e.g. CoreOS or GCI) that have chosen not to include iscsiadm in their defualt install.

Should the list include all tests that should work in *our* CI, where we can can assume particular RHEL version and limited set of cloud providers? Or should it include all tests that should work on a *generic* Linux distro?